activerecord/test/cases/relations_test.rb

```diff
@@ -182,6 +182,42 @@ def test_finding_with_subquery_without_select_does_not_change_the_select
     end
   end
 
+  def test_select_with_original_table_name_in_from
+    relation = Comment.joins(:post).select(:id).order(:id)
+    subquery = Comment.from(Comment.table_name).joins(:post).select(:id).order(:id)
+    assert_equal relation.map(&:id), subquery.map(&:id)
+  end
+
+  def test_pluck_with_original_table_name_in_from
+    relation = Comment.joins(:post).order(:id)
+    subquery = Comment.from(Comment.table_name).joins(:post).order(:id)
+    assert_equal relation.pluck(:id), subquery.pluck(:id)
+  end
+
+  def test_select_with_quoted_original_table_name_in_from
+    relation = Comment.joins(:post).select(:id).order(:id)
+    subquery = Comment.from(Comment.quoted_table_name).joins(:post).select(:id).order(:id)
+    assert_equal relation.map(&:id), subquery.map(&:id)
+  end
+
+  def test_pluck_with_quoted_original_table_name_in_from
+    relation = Comment.joins(:post).order(:id)
+    subquery = Comment.from(Comment.quoted_table_name).joins(:post).order(:id)
+    assert_equal relation.pluck(:id), subquery.pluck(:id)
+  end
+
+  def test_select_with_subquery_in_from_uses_original_table_name
+    relation = Comment.joins(:post).select(:id).order(:id)
+    subquery = Comment.from(Comment.all, Comment.quoted_table_name).joins(:post).select(:id).order(:id)
+    assert_equal relation.map(&:id), subquery.map(&:id)
```

Comment:
Weird... This test isn't failed locally... I suspect a bug on SQLite3...

https://travis-ci.org/rails/rails/jobs/496726410#L1198-L1208

Reply:
It did fail on buildkite as well https://buildkite.com/rails/rails/builds/58981

Reply:
ah... skipped on buildkite as well for now a333ba3f7f24dff3c50c7cf26a2fe2caa748a851.